anker/src/processor.rs

```diff
@@ -327,6 +329,19 @@ fn process_sell_rewards(program_id: &Pubkey, accounts_raw: &[AccountInfo]) -> Pr
         accounts.anker.key,
         accounts.st_sol_reserve_account,
     )?;
+
+    let clock = Clock::from_account_info(accounts.sysvar_clock)?;
+    let most_recent_sample = anker.historical_st_sol_prices.last();
```

Comment:
The max age should apply to the _least recent sample_, not to the most recent one! Otherwise, you can have _n_ - 1 very old samples, add one new sample, and then already call `SellRewards`. We need to compare against `historical_st_sol_prices[0]`.

Anyway, did I not implement this already? I thought I did. Maybe I didn’t push it :thinking:

Reply:
Humm... That's what I was wondering. It's better like this, I was going to add another check for ensuring the past values were done in < minimum, but if we check the first one we ensure the newer entries are correct.